src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -170,6 +174,20 @@ public IStringLocalizer Create(string baseName, string location)
             });
         }
 
+        protected virtual string GetResourcePrefix(string location, string baseName, string resourceLocation)
+        {
+            // Re-root the base name if a resources path is set
+            return location + "." + resourceLocation + TrimPrefix(baseName, location + ".");
+        }
+        private string GetResourcePath(Assembly assembly)
+        {
+            var resourceLocationAttribute = assembly.GetCustomAttribute<ResourceLocationAttribute>();
```

Comment:
To allow unit testing, need a method or two which has a `ResourceLocationAttribute` parameter and is available from test assembly. For example, confirm correct resource path is chosen if `ResourceLocation` contains directory separators.